### ISSUE TITLE
"gsl.h" requires <string> 

### ISSUE THREAD

**Issue by rmpowell77** [Microsoft/GSL#235]
The following fails with clang-700.1.81:

```
#include "gsl.h"
int main(int argc, char* argv[])
{}
```
Fails with this error
```
In file included from opensource/GSL/include/gsl.h:25:
opensource/GSL/include/string_span.h:290:38: error: implicit instantiation of undefined
      template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
        : span_(const_cast<pointer>(s.data()), narrow_cast<std::ptrdiff_t>(s.length()))
                                     ^
...
```

Looks like string_span.h was expecting string to be included.  If I change to:
```
#include <string>
#include "gsl.h"

int main(int argc, char* argv[])
{}
```

Compiles fine.  Perhaps string_span.h should include string?

**Comment by neilmacintosh** [comment]
Thanks @rmpowell77 for the report! This should now have been fixed by PR #233.

**Comment by rmpowell77** [comment]
LGTM
